src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAction.cs

```diff
@@ -25,42 +25,85 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.IdentityModel.Logging;
+using System;
+using static Microsoft.IdentityModel.Logging.LogHelper;
 
 namespace Microsoft.IdentityModel.Tokens.Saml
 {
+    /// <summary>
+    /// Represents the Action element specified in [Saml, 2.4.5.1].
+    /// </summary>
     public class SamlAction
     {
-        string _ns;
-        string _action;
+        private Uri _namespace;
+        private string _value;
 
+        /// <summary>
+        /// Constructs an instance of SamlAction class.
+        /// </summary>
         internal SamlAction() { }
 
+        /// <summary>
+        /// Constructs an instance of Saml2Action class.
```

Comment:
We should be using <see cref="SamlAction"> I have asked Valentin to pick this up. Cause it seems all over.